source/game_sa/Hud.cpp

```diff
@@ -465,12 +504,150 @@ void CHud::DrawAfterFade() {
 
 // 0x58AA50
 void CHud::DrawAreaName() {
-    plugin::Call<0x58AA50>();
+    if (!m_pZoneName) {
+        return;
+    }
+
+    
+    if (m_pZoneName != m_pLastZoneName) {
+        switch (m_ZoneState) {
+        case NAME_DONT_SHOW:
+            if (!CTheScripts::bPlayerIsOffTheMap && CTheScripts::bDisplayHud ||
+                CEntryExitManager::ms_exitEnterState == 1 ||
+                CEntryExitManager::ms_exitEnterState == 2) {
+                m_ZoneState = NAME_FADE_IN;
+                m_ZoneNameTimer = 0;
+                m_ZoneFadeTimer = 0;
+                m_ZoneToPrint = m_pZoneName;
+                if (m_VehicleState == NAME_SHOW ||
+                    m_VehicleState == NAME_FADE_IN)
+                    m_VehicleState = NAME_FADE_OUT;
+            }
+            break;
+        case NAME_SHOW:
+        case NAME_FADE_IN:
+        case NAME_FADE_OUT:
+            m_ZoneState = NAME_SWITCH;
+            m_ZoneNameTimer = 0;
```

Comment:
Missing `break` maybe? Please check.
```suggestion
            m_ZoneNameTimer = 0;
            break;
```